src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -262,11 +290,20 @@ public void validate() throws AppEngineException {
           "Validation Error: Java Tools jar location '"
               + JAR_LOCATIONS.get(JAVA_TOOLS_JAR) + "' is not a file.");
     }
+    try {
+      if (!isComponentInstalled("app-engine-java")) {
```

Comment:
The practice of depending on the Cloud SDKs internal directory structure is not OK with the Cloud SDK team.  They reserve the right to break this at their discretion.  We need to move away from depending on incidental files distributed with the SDK and moving to using their api/cli for this check is part of that.

This is also why we're moving to using Maven central as the source for classpath deps.